common/src/main/java/net/william278/huskhomes/command/TpCommand.java

```diff
@@ -88,9 +87,24 @@ private void execute(@NotNull CommandUser executor, @NotNull Teleportable telepo
         final TeleportBuilder builder = Teleport.builder(plugin)
                 .teleporter(teleportable)
                 .target(target);
-        if (executor instanceof OnlineUser user) {
-            builder.executor(user);
+
+        if (executor instanceof OnlineUser onlineUser) {
+            if (target instanceof Teleportable teleportableTarget) {
+                if (!onlineUser.hasPermission(getPermission("other"))) {
```

Comment:
This logic is _still_ wrong - if `onlineUser` and `teleportableTarget` are the same person, this check will still incorrectly be triggered.